lib/hanami/entity/schema.rb

```diff
@@ -93,11 +93,22 @@ class Definition
         #
         # @since 0.7.0
         class Dsl
+          # @since x.x.x
+          # @api private
+          TYPES = [:schema, :strict, :weak, :permissive, :strict_with_defaults, :symbolized].freeze
```

Comment:
I'm a little wary of copying these values from `dry-types`. I'm not sure there's a better solution though. If they add some new contstructor_type, it'd be nice to support it automatically.

`Dry::Types::Hash.public_instance_methods.first(6)` returns this same thing, but in a different order. This depends on their ordering of methods, which would be quite brittle, unfortunately :/